test/fixtures/rules/std_rule_cases/L049.yml

```diff
@@ -6,6 +6,14 @@ test_is_null:
     FROM foo
     WHERE a IS NULL
 
+test_equal_string:
+  pass_str: |
+    SELECT
+        foo
+    FROM restricted.tble
+    WHERE country = 'DE' AND ip_hash NOT NULL
```

Comment:
Yea Looking back Im quite confused. I can say these things for certain
1. It parses
2. Its multiple times in my running[?] code base (Redshift)
3. I cant get a simple version of it to run vs Redshift

Reply:
Is there an easy way to figure out how and why this is parsing?

Reply:
It's parsing because of this:

https://github.com/sqlfluff/sqlfluff/blob/9a408d3798d1257a20943b4a04ebd6d10c914956/src/sqlfluff/dialects/dialect_ansi.py#L1649-L1654

The Expression Grammar is quite complex and reused anywhere there's an expression meaning it might let things like this through when perhaps it should not.

Weird that you have it in your code but can't get it to run?

Reply:
Sadly Not weird we've 1000s of file and 70 people writing them - Surprising that most of them run.

Reply:
Sounds like you need a good SQL Linter to catch at least some of these! 😉

Anyway, I presume this is still a worthwhile change since technically SQLFluff didn't know this was bad SQL and thought it was good. and still corrupted it?

Reply:
Lol yea. I guess this PR still makes sense - cause it will prevent totally destroying the SQL file.
and this rewrite also fixes things into the fn style (which seems to be the new one), and I think makes the logic here more correct in general (the rule was still too eager).

Reply:
Anyway, think we should remove this test case as not valid SQL (even if SQLFluff doesn't know it).

Reply:
I agree with removing it. Is there a similar test case (i.e. that would test the same case) that **is** valid SQL?